receiver/prometheusreceiver/config.go

```diff
@@ -110,17 +113,30 @@ func (cfg *Config) Unmarshal(componentParser *configparser.Parser) error {
 		return fmt.Errorf("prometheus receiver failed to parse config: %s", err)
 	}
 
-	// Unmarshal prometheus's config values. Since prometheus uses `yaml` tags, so use `yaml`.
-	promCfg, err := componentParser.Sub(prometheusConfigKey)
-	if err != nil || len(promCfg.ToStringMap()) == 0 {
-		return err
-	}
-	out, err := yaml.Marshal(promCfg.ToStringMap())
-	if err != nil {
-		return fmt.Errorf("prometheus receiver failed to marshal config to yaml: %s", err)
+	var yamlBlob []byte
+
+	// TODO: Ensure that only 1 of "config_file" or the configuration exist.
+
+	switch cfg.File == "" {
+	case true: // No file was provided, the common case, parse as normally as we did before.
```

Comment:
nit: readers won't know the history, so try not to reference what "we did before" in comments.

Reply:
Removed, thank you!